### PR TITLE
refactor(Metrics): filter out "bad" metrics

### PR DESCRIPTION
**Pull request by Yimura**
Thinking about adding the "bad metric" log behind a toggle to prevent users from worrying about the log.

### CODE REVIEW

File: src/hooks/info/prepare_metric_for_sending.cpp [hunk 1]
```diff
@@ -3,50 +3,53 @@
...
+	    "CF",
+	    "CC",
+	    "CNR",
+	    "SCRIPT",
+	    "AUX_DEUX",
+	    "HARDWARE_OS",
+	    "HARDWARE_GPU",
+	    "HARDWARE_MOBO",
+	    "HARDWARE_MEM",
+	    "HARDWARE_MEM",
```
Comment (tupoy-ya) [inline src/hooks/info/prepare_metric_for_sending.cpp]: Duplicate `HARDWARE_MEM` entry.

Comment (Yimura) [reply src/hooks/info/prepare_metric_for_sending.cpp]: I will adjust the list with a couple more ones when I get home.

**Comment by gir489returns** [comment]
Will this fix detections?

**Comment by Yimura** [comment]
> Will this fix detections?

No but it's just improving old code so why not.

**Comment by gir489returns** [comment]
> > Will this fix detections?
> 
> No but it's just improving old code so why not.

The title is misleading.

**Comment by Yimura** [comment]
> The title is misleading.

Well it sorta says what it does without the "allow normal metrics to go through".

**Comment by gir489returns** [comment]
> > The title is misleading.
> 
> Well it sorta says what it does without the "allow normal metrics to go through".

I'm confused. Are we sending the normal metrics or not?

**Comment by Yimura** [comment]
> I'm confused. Are we sending the normal metrics or not?

Yes with this PR we're allowing normal ones and filtering out "bad" ones.

**Pull request merged**